source/bdDetect.py

```diff
@@ -210,6 +225,24 @@ class _DeviceInfoFetcher(AutoPropertyObject):
 	"""Utility class that caches fetched info for available devices for the duration of one core pump cycle."""
 	cachePropertiesByDefault = True
 
+	def __init__(self):
+		self._btDevsLock = threading.Lock

```

Comment:
This should create the lock here, usages should be `self._btDevsLock` not `self._btDevsLock()`

```suggestion
		self._btDevsLock = threading.Lock()
```